.github/workflows/test.yml

```diff
@@ -1,6 +1,10 @@
 name: test
 
-on: workflow_dispatch
+on: [push]
+
+concurrency:
+  group: ${{github.workflow}}-${{github.ref}}
+  cancel-in-progress: true
```

Comment:
Why cancel runs for other PRs?

Reply:
this section just makes sure that if someone is pushing several commits to the same branch/PR then it will cancel the one that's already in-progress and run just the latest one. I think it's useful if someone is developing on a specific PR for example and they push 3 commits in 2 minutes then it will make sure that only the latest job will be run to completion. do you think it's better to take it out?